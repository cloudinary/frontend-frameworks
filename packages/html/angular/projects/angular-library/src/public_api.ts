/*
 * Public API Surface of angular-library
 */

export * from './lib/angular-library.service';
export * from './lib/angular-library.component';
export * from './lib/angular-library.module';
