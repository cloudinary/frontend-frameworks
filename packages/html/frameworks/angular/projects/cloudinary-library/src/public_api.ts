/*
 * Public API Surface of cloudinary-library
 */

import {
  placeholder,
  accessibility,
  lazyload,
  responsive,
} from '../../../../html/dist';
export {placeholder, accessibility, lazyload, responsive};

export * from './lib/cloudinary.module';
