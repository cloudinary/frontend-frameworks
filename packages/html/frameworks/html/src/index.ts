export { HtmlImageLayer } from './layers/htmlImageLayer';
export { HtmlVideoLayer } from './layers/htmlVideoLayer';
export { HtmlPictureLayer } from './layers/htmlPictureLayer';
export {responsive} from './plugins/responsive';
export {lazyload} from './plugins/lazyload';
export {accessibility} from './plugins/accessibility';
export {placeholder} from './plugins/placeholder';
export {isBrowser} from './utils/isBrowser';
export {serverSideSrc} from './utils/serverSideSrc';
export {Plugins, VideoSources, PictureSources} from './types';
export {cancelCurrentlyRunningPlugins} from './utils/cancelCurrentlyRunningPlugins';
