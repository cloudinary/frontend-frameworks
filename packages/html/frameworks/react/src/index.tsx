import {
  placeholder,
  accessibility,
  lazyload,
  responsive,
} from '../../html/dist'
import {AdvancedImage} from "./AdvancedImage";


export {placeholder, accessibility, lazyload, responsive, AdvancedImage};
