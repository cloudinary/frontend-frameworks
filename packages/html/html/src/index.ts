export { HtmlLayer } from './htmlLayer';
export {responsive} from './responsive';
export {lazyload} from './lazyload';
export {accessibility} from './accessibility';
export {placeholder} from './placeholder';
export {isBrowser} from './isBrowser';
export {serverSideSrc} from './serverSideSrc';
export {html} from './html'
export {plugins} from './htmlLayer';

