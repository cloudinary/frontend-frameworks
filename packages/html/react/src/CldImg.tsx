import React from 'react';
import {CloudinaryImage} from "@cloudinary/base/assets/CloudinaryImage";

import {
  HtmlLayer,
  plugins
} from '../../html/dist'

interface ImgProps {
  transformation: CloudinaryImage,
  plugins?:  plugins,
  [x: string]: any
}

export class CldImg extends React.Component <ImgProps> {
  imageRef: React.RefObject<HTMLImageElement>;
  htmlLayerInstance: HtmlLayer;

  constructor(props: ImgProps) {
    super(props);
    this.imageRef = React.createRef();
  }

  /**
   * On mount creates a new HTMLLayer instance and initialises with ref to img element,
   * user generated cloudinaryImage and the plugins to be used
   */
  componentDidMount() {
    this.htmlLayerInstance = new HtmlLayer(
      this.imageRef.current,
      this.props.transformation,
      this.props.plugins,
    )
  }

  /**
   * On update we cancel running plugins and update image instance with the state of user
   * cloudinaryImage and the state of plugins
   */
  componentDidUpdate() {
    this.htmlLayerInstance.cancelCurrentlyRunningPlugins();
    // call html layer to update the dom again with plugins and reset toBeCanceled
    this.htmlLayerInstance.update(this.props.transformation, this.props.plugins)
  }

  /**
   * On unmount we cancel the currently running plugins
   */
  componentWillUnmount() {
    // safely cancel running events on unmount
    this.htmlLayerInstance.cancelCurrentlyRunningPlugins()
  }

  render() {
    const {
      transformation,
      plugins,
      ...otherProps // Assume any other props are for the base element
    } = this.props;
    return <img {...otherProps} ref={this.imageRef} />
  }
}
