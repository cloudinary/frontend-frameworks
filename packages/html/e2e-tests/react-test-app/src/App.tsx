import React, {Fragment, useState} from 'react'
import {AdvancedImage, responsive, placeholder} from '@cloudinary/react';
import {CloudinaryImage} from "@cloudinary/url-gen/assets/CloudinaryImage";

const tests = [
  'responsive',
  'placeholder',
  'lazy',
  'lazyPlaceholder',
  'lazyResponsive',
  'responsivePlaceholder'
];

const App = () => {
  let img = new CloudinaryImage('sample', {cloudName: 'demo'}, { analytics: false });
  const [test, setTest]: [any, any] = useState(0);

  const Buttons = () => (
    <Fragment>
      {
        tests.map((t, i) =>
          <button key={"btn-" + i} id={t + 'Btn'} onClick={() => setTest(t)}>{t + ' test'}</button>
        )
      }
    </Fragment>
  );

  return (
    <div id="top">
      <Buttons/>
      {test === 'responsive' &&
      <Fragment>
        <h1>Responsive Image</h1>
        <div style={{width: "330px"}}>
          <AdvancedImage id="responsive" cldImg={img} plugins={[responsive(100)]}/>
        </div>
      </Fragment>
      }
      {test === 'placeholder' &&
      <div>
        <h1>Placeholder</h1>
        <AdvancedImage id="placeholder" cldImg={img} plugins={[placeholder()]}/>
      </div>
      }
    </div>
  );
}

export default App;

