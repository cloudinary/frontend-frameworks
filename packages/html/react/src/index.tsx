import * as React from 'react'
import {
  HtmlLayer,
  placeholder,
  accessibility,
  lazyload,
  responsive,
  cancelCurrentlyRunningPlugins,
  isBrowser,
  serverSideSrc
} from '../../html/dist'

export class CldImg extends React.Component {
  private myRef: React.RefObject<unknown>;
  private canceled: any[];
  private imageInstance: HtmlLayer;
  constructor(props: any) {
    super(props);
    this.myRef = React.createRef();
    this.state = {
      transformation: props.transformation,
      plugins: props.plugins
    }
  }

  /**
   * On mount creates a new HTMLLayer instance and initialises with ref to img element,
   * user generated transformableImg, the plugins to be used, and an empty array to hold state
   * of cancelled plugins
   */
  componentDidMount() {
    this.canceled = []; // holds 'canceled' resolves
    this.imageInstance = new HtmlLayer(
      this.myRef.current,
      // @ts-ignore
      this.state.transformation,
      // @ts-ignore
      this.state.plugins,
      this.canceled
    )
  }

  /**
   * On update we cancel running plugins and update image instance with the state of user
   * transformableImg, the state of plugins, and we reset the state of the canceled plugins
   */
  componentDidUpdate() {
    cancelCurrentlyRunningPlugins(this.canceled);
    // call render again with plugins and reset toBeCanceled
    // @ts-ignore
    this.imageInstance.update(this.state.transformation, this.state.plugins, [])
  }

  /**
   * On unmount we cancel the currently running plugins
   */
  componentWillUnmount() {
    // safely cancel running events on unmount
    cancelCurrentlyRunningPlugins(this.canceled)
  }

  render() {
    if (isBrowser()) {
      console.log('client');
      // @ts-ignore
      return <img suppressHydrationWarning {...this.props} ref={this.myRef} />
    } else {
      console.log('server');
      const src = serverSideSrc(
        '', // src
        // @ts-ignore
        this.state.plugins,
        // @ts-ignore
        this.props.transformation
      );
      return <img {...this.props} src={src} />
    }
  }
}

export {placeholder, accessibility, lazyload, responsive};

