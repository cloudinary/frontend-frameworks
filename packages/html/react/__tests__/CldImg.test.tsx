import { CldImg } from '../src'
import {CloudinaryImage} from "@cloudinary/base/assets/CloudinaryImage";
import CloudinaryConfig from "@cloudinary/base/config/CloudinaryConfig";
import {mount} from 'enzyme';
import React from "react";

const CONFIG_INSTANCE = new CloudinaryConfig({
  cloud: {
    cloudName: 'demo'
  }
});

let cl = new CloudinaryImage('sample').setConfig(CONFIG_INSTANCE);


describe('CldImg', () => {
  it('is truthy', () => {
    expect(CldImg).toBeTruthy()
  });

  it('renders CldImg', async () => {
    const component = await mount(<CldImg transformation={cl}/>);
    expect(component).toMatchSnapshot();
  });

  it("should create an img tag", async function() {
    let component = await mount(<CldImg transformation={cl}/>);
    expect(component.html()).toBe('<img src="https://res.cloudinary.com/demo/image/upload/sample">');
  });

  it("should add style to img component", async function() {
    let component = await mount(<CldImg style={{opacity: "0.5"}} transformation={cl}/>);
    expect(component.find('img').prop('style')).toStrictEqual({"opacity": "0.5"});
  });
});
