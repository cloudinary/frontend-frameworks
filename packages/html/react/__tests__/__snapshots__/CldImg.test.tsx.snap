// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CldImg renders CldImg 1`] = `ReactWrapper {}`;
